File: mobile/lib/core/models/exec_sp_response.dart
class ExecSpResponse {
  final bool success;
  final dynamic data;
  final String? error;

  ExecSpResponse({
    required this.success,
    this.data,
    this.error,
  });

  factory ExecSpResponse.fromJson(Map<String, dynamic> json) => ExecSpResponse(
        success: json['success'] as bool? ?? json['Success'] as bool? ?? false,
        data: json['data'] ?? json['Data'],
        error: json['error'] as String? ?? json['Error'] as String?,
      );
}

File: api/CavaliERP.API/Models/ExecSpResponse.cs
namespace CsmStok.Api.Models;

public sealed class ExecSpResponse
{
    public bool Success { get; init; }
    public object? Data { get; init; }
    public string? Error { get; init; }

    public static ExecSpResponse Ok(object? data) => new() { Success = true, Data = data };
    public static ExecSpResponse Fail(string error) => new() { Success = false, Error = error };
}
